orchestra/todos/views.py

```diff
@@ -178,3 +118,78 @@ class TodoListTemplateList(generics.ListCreateAPIView):
 
     serializer_class = TodoListTemplateSerializer
     queryset = TodoListTemplate.objects.all()
+
+
+class GenericTodoViewset(ModelViewSet):
+    serializer_class = BulkTodoSerializer
+    filter_backends = (filters.DjangoFilterBackend,)
+    filterset_fields = ('project', 'step',)
+    queryset = Todo.objects.all()
+
+    def get_serializer(self, *args, **kwargs):
+        if isinstance(kwargs.get('data', {}), list):
+            kwargs['many'] = True
+
+        return super().get_serializer(*args, **kwargs)
+
+    def get_queryset(self, ids=None):
+        queryset = super().get_queryset()
+        if ids:
+            queryset = queryset.filter(id__in=ids)
+        return queryset.order_by('-created_at')
+
+    @action(detail=False, methods=['put'])
+    def put(self, request, *args, **kwargs):
+        ids = [x['id'] for x in request.data]
+        instances = self.get_queryset(ids=ids)
+        serializer = self.get_serializer(
+            instances, data=request.data, partial=False, many=True)
+        serializer.is_valid(raise_exception=True)
+        self.perform_update(serializer)
+        data = serializer.data
+        return Response(data)
+
+    def _get_sender(self):
+        if isinstance(self.request.user, SignedUser):
+            return None
+        return Worker.objects.get(
+            user=self.request.user).formatted_slack_username()
+
+    def perform_update(self, serializer):
+        if isinstance(serializer.validated_data, list):
+            serializer.save()
```

Comment:
in both branches we call serializer.save, so consider calling that before branching and having a single `if`